src/openms/include/OpenMS/ANALYSIS/OPENSWATH/ChromatogramExtractor.h

```diff
@@ -317,25 +311,46 @@ namespace OpenMS
           prec.setIsolationWindowUpperOffset(settings.getPrecursors()[0].getIsolationWindowUpperOffset());
         }
 
-        // 3) set precursor peptide sequence / compound id in more-or-less default field
-        String pepref = transition->getPeptideRef();
-        for (Size pep_idx = 0; pep_idx < transition_exp.getPeptides().size(); pep_idx++)
+        if (transition->getTransType() == OpenSwath::TransType::PEPTIDE)
         {
-          const OpenMS::TargetedExperiment::Peptide* pep = &transition_exp.getPeptides()[pep_idx];
-          if (pep->id == pepref)
+          // 3) set precursor peptide sequence / compound id in more-or-less default field
+          String pepref = transition->getTransRef();
+          for (Size pep_idx = 0; pep_idx < transition_exp.getPeptides().size(); pep_idx++)
           {
-            prec.setMetaValue("peptide_sequence", pep->sequence);
-            break;
+            const OpenMS::TargetedExperiment::Peptide* pep = &transition_exp.getPeptides()[pep_idx];
+            if (pep->id == pepref)
+            {
+              prec.setMetaValue("peptide_sequence", pep->sequence);
+              break;
+            }
           }
         }
-        String compref = transition->getCompoundRef();
-        for (Size comp_idx = 0; comp_idx < transition_exp.getCompounds().size(); comp_idx++)
+        else if (transition->getTransType() == OpenSwath::TransType::NUCTIDE)
         {
-          const OpenMS::TargetedExperiment::Compound* comp = &transition_exp.getCompounds()[comp_idx];
-          if (comp->id == compref)
+          // 3) set precursor nuctide sequence / compound id in more-or-less default field
+          String nucref = transition->getTransRef();
+          for (Size nuc_idx = 0; nuc_idx < transition_exp.getNuctides().size(); nuc_idx++)
           {
-            prec.setMetaValue("peptide_sequence", String(comp->id) );
-            break;
+            const OpenMS::TargetedExperiment::Nuctide* nuc = &transition_exp.getNuctides()[nuc_idx];
+            if (nuc->id == nucref)
+            {
+              prec.setMetaValue("peptide_sequence", nuc->id);
```

Comment:
Should this metavalue be renamed to something more encompassing? possibly transition_group_name?

Reply:
This specific instance gets nuked with removal of the deprecated code. But I will update it other places.